### DISCUSSION TITLE
Enable DuckDB Extensions for connector

### DISCUSSION THREAD

**Discussion by archiewood** [evidence-dev/evidence#2357] [Data Source Request]
See #644 

HTTPFS is allows you to query CSV / parquet files that are stored in S3 using DuckDB

## Implementation ideas
- Checkbox in the settings UI where you connect to DuckDB ("Use HTTPFS" or similar), which then shows fields to enter of `s3_region` and `s3_access_key_id`, and `s3_secret_key_id`.
- Use the duckdb connector to conditionally run the required SQL statements, if HTTPFS is selected?

**Comment by archiewood** [comment]
It would also be good to enable https://duckdb.org/docs/guides/import/excel_import.html for users.

**Comment by hydrojumbo** [comment]
JSON support would make it easier to handle JSON columns in queries: https://duckdb.org/docs/extensions/json.html#installing-and-loading. For now, this can be handled in the sources query if the source database supports it.

**Comment by archiewood** [comment]
Support for DuckDB extension autoloading shipped with DuckDB 1.0, so any of the default library of extensions should work in DuckDB source queries.

Other extension not yet supported
